linkup-cli/src/background_booting.rs

```diff
@@ -31,12 +29,16 @@ pub fn boot_background_services() -> Result<(), CliError> {
 
     wait_till_ok(format!("{}linkup-check", local_url))?;
 
-    if is_tunnel_started().is_err() {
+    if state.should_use_tunnel() && is_tunnel_started().is_err() {
         println!("Starting tunnel...");
         let tunnel = start_tunnel()?;
-        state.linkup.tunnel = tunnel;
-    } else {
+        state.linkup.tunnel = Some(tunnel);
+    } else if state.should_use_tunnel() {
```

Comment:
This condition is repeated twice; I think you could check for it once and move the other condition inside that If-statement as it's a little hard to follow the logic.